ui/apps/platform/src/Containers/Dashboard/PatternFly/ScopeBar.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useMemo } from 'react';
+import { Toolbar, ToolbarContent, ToolbarItem } from '@patternfly/react-core';
+import { gql, useQuery } from '@apollo/client';
+import omit from 'lodash/omit';
+
+import useURLSearch from 'hooks/useURLSearch';
+
+import { flattenFilterValue } from 'utils/searchUtils';
+import NamespaceSelect from './NamespaceSelect';
+import ClusterSelect, { SelectionChangeAction } from './ClusterSelect';
+import { Cluster } from './types';
+
+type NamespacesResponse = {
+    clusters: Cluster[];
+};
+
+const namespacesQuery = gql`
+    query getAllNamespacesByCluster($query: String) {
+        clusters(query: $query) {
+            id
+            name
+            namespaces {
+                metadata {
+                    id
+                    name
+                }
+            }
+        }
+    }
+`;
+
+function ScopeBar() {
+    const { searchFilter, setSearchFilter } = useURLSearch();
+    const { data, loading, error } = useQuery<NamespacesResponse>(namespacesQuery, {
+        variables: { query: '' },
+    });
+    const selectedClusterData = useMemo(() => {
+        return data?.clusters.filter(({ name }) => searchFilter.Cluster?.includes(name)) ?? [];
```

Comment:
This searching an array for values in another array isn't great, but I'm not sure that number of items we are comparing here calls for making this more complicated/performant.